### ISSUE TITLE
Wrapping dylib into a framework

### ISSUE THREAD

**Issue by Davvie** [apple/turicreate#2915]
Unfortunately wrapping `libRecommender.dylib` into a framework doesn't seem to help with the App Store delivery error:
![Снимок экрана 2020-01-13 в 3 00 06](https://user-images.githubusercontent.com/816045/72227716-ede5d480-35b0-11ea-8f63-fc33ee160ca9.png)

![Снимок экрана 2020-01-13 в 3 01 35](https://user-images.githubusercontent.com/816045/72227754-2a193500-35b1-11ea-83e4-2b624f50ace8.png)

Does anyone else experience this issue?

**Comment by TobyRoseman** [comment]
This is related to #2050.

@Davvie - Are you following the [directions from our User Guide](https://apple.github.io/turicreate/docs/userguide/recommender/coreml-deployment.html)? Specifically, are you following the instructions from the section which begins "If your project will be submitted to the App Store"?

**Comment by Davvie** [comment]
@TobyRoseman Yes, I am. I guess I've tried all possible combinations with and without the libAudioPreprocessing.dylib in my framework.
Could you kindly provide an empty sample project with everything set up the right way? It would help a lot to find the problem in my configuration.

**Comment by syoutsey** [comment]
@Davvie Can you share your project so we can take a look? Also, are you using the `libRecommender`, the `libAudioPreprocessing`, or both? If both, is there a chance that you've correctly configured one framework but not the other?

**Comment by Davvie** [comment]
@syoutsey I'm only using `libRecommender` so I didn't include `libAudioPreprocessing` in my first attempts. Should I include it anyway?

Here is an empty project with the project configuration similar to mine: https://www.dropbox.com/s/0hl19wnm15g668q/RecommenderSample.zip?dl=0
If you could take a look at it I'd be really grateful

**Comment by FamilGadzhiev** [comment]
Hi! I'm having the same issue, can't submit my app to the App Store :(
@davvie checked out your link, I have the same setup

**Comment by FamilGadzhiev** [comment]
> @Davvie Can you share your project so we can take a look? Also, are you using the `libRecommender`, the `libAudioPreprocessing`, or both? If both, is there a chance that you've correctly configured one framework but not the other?

Hi! I'm having the same issue, can't submit my app to the App Store :(
@davvie checked out your link, I have the same setup
could u help? @syoutsey

**Comment by FamilGadzhiev** [comment]
After many attempts I gave up trying to wrap the dylib into a framework and tried to include source files  from [this folder](https://github.com/apple/turicreate/tree/master/src/deployment) manually.

Still no success though, now the Suggestions class can not be found. Am I missing something? :(

**Comment by Davvie** [comment]
**Update:** I finally managed to submit a build!
The trick was to select "Objective-C" as a language when creating the framework target

**Comment by Davvie** [comment]
Hi, I guess I haven't solved the issue even though the submission passed successfully. The app crashes on launch because the library can't be found in the configuration I made in the end.

I emailed the developer technical support, and they said that "there is no supported way to take a pre-build .dylib file and package it as a framework".

@TobyRoseman @syoutsey  Is there any other way I can import the recommender library other than the dylib? In case of the sound classifier I can just drag and drop the .h and .m files from the 'src/deployment' folder.

**Comment by TobyRoseman** [comment]
We are exploring another option in #3023.

**Comment by Davvie** [comment]
Sounds great, thank you! Looking forward to it

**Comment by davidesoldan** [comment]
> I emailed the developer technical support, and they said that "there is no supported way to take a pre-build .dylib file and package it as a framework".

Hi @Davvie 
sorry for the stupid question. But that means that is not possible to use a .dylib inside an iOS application in any way? I mean, for iOS apps that goes to the App Store / TestFlight?

From the section "Embedded .dylib Files" of the documentation here
[https://developer.apple.com/library/archive/technotes/tn2435/_index.html#//apple_ref/doc/uid/DTS40017543-CH1-TROUBLESHOOTING-BUNDLE_ERRORS](url)
I know that I cannot use directly a .dylib file into an app, but I have to wrap it into a target framework.
But the developer support tells you that there is no supported way to do it.
I'm a little confused
Thank you

**Comment by anhar** [comment]
I would love to get an update on this as I have the exact same problem moving the `libAudioPreprocessing.dylib` into its own framework. Compiling and installation on device works fine but during runtime before `didFinishLaunchingWithOptions` is called the iOS application crashes every time:

```
dyld: Library not loaded: /Users/zach/builds/1fUgszd-/0/turi/turicreate/release/src/deployment/Release-iphoneos/libAudioPreprocessing.dylib
  Referenced from: /private/var/containers/Bundle/Application/D4308FD3-DFCD-4596-B23A-6FF9AA9262B7/Foo.app/Frameworks/Bar.framework/AudioPreprocessing
  Reason: image not found
```
